### PR TITLE
add <title>xyz</title> tag to <svg> icons in order to confirm to W3C accessibility standards

### PR DESCRIPTION
**Pull request by vbence86**
### Disclaimer
Accessibility guidelines required that SVG graphics have a <title></title> child element that explains its meaning. The icon taglib(s) that produce SVGs should at least add the icon label within a child title element.

https://www.w3.org/TR/SVG/access.html

Steps
1. Navigate browser to a Liferay page with the Documents and Media portlet
2. Inspect an icon element on the portlet
3. Look for title child element within the svg icon element
Expected: A title element exists that explains the meaning of the icon
Result: No title element exists that explains the meaning of the icon

### Intent
To update modules that output `<svg>` tags in order to render the required `<title>...</title>` so that the output HTML conforms to W3C standards with respect to accessibility.

### Solution
- Carefully went through the code base chasing down all occurrences of `<svg>` creation. 
- Updated these sections to also render `<title>...</title>` child elements

#### Test
- Tested against my local liferay build and checked whether all `<svg>` tags contain `<title>...</title>` additions.

lps-68726

### CODE REVIEW

File: modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ AUI.add(
 									{
 										cssClass: 'close',
 										discardDefaultButtonCssClasses: true,
-										labelHTML: '<svg class="lexicon-icon" focusable="false"><use data-href="' + Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg#times" /></svg>',
+										labelHTML: '<svg class="lexicon-icon" focusable="false"><use data-href="' + Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg#times" /><title>Close</title></svg>',
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util_window.js]: Title shouldn't be localized?

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/diff_version_comparator/js/diff_version_comparator.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ AUI.add(
 
 		var TPL_VERSION = '<div class="pull-right close-version-filter">' +
 				'<a href="javascript:;">' +
-					'<svg class="lexicon-icon" focusable="false"><use data-href="' + Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg#times" /></svg>' +
+					'<svg class="lexicon-icon" focusable="false"><use data-href="' + Liferay.ThemeDisplay.getPathThemeImages() + '/lexicon/icons.svg#times" /><title>Icon</title></svg>' +
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/diff_version_comparator/js/diff_version_comparator.js]: Title should be localized text "Close" instead Icon

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy [hunk 3]
```diff
@@ -88,6 +88,7 @@
 					<span class="icon-monospaced" data-onclick="{$removeAvailableLocale}" data-locale-id="{$locale.id}" href="javascript:;">
 						<svg class="lexicon-icon">
 							<use xlink:href="{$pathThemeImages}/lexicon/icons.svg#times"></use>
+							<title>Icon</title>
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy]: Title should be localized text "Close"

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy [hunk 4]
```diff
@@ -113,6 +114,7 @@
 				<span class="icon-monospaced">
 					<svg class="lexicon-icon">
 						<use xlink:href="{$pathThemeImages}/lexicon/icons.svg#{$locale.icon}"></use>
+						<title>Icon</title>
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy]: I guess title here should be {$locale.icon}

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy [hunk 5]
```diff
@@ -141,6 +143,7 @@
 			<a class="icon-monospaced" data-onclick="toggle" href="javascript:;">
 				<svg class="lexicon-icon">
 					<use xlink:href="{$pathThemeImages}/lexicon/icons.svg#{$icon}"></use>
+					<title>Icon</title>
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy]: I guess title here should be {$locale.icon}

File: modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy [hunk 6]
```diff
@@ -153,6 +156,7 @@
 						<span class="icon-monospaced">
 							<svg class="lexicon-icon">
 								<use xlink:href="{$pathThemeImages}/lexicon/icons.svg#{$locale.icon}"></use>
+								<title>Icon</title>
```
Comment (carloslancha) [inline modules/apps/foundation/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/translation_manager/TranslationManager.soy]: I guess title here should be {$locale.icon}

File: modules/apps/foundation/portal-search/portal-search-web/src/main/resources/META-INF/resources/search/results/view.jsp [hunk 7]
```diff
@@ -99,6 +99,7 @@ com.liferay.portal.kernel.dao.search.SearchContainer<com.liferay.portal.kernel.s
 						<span class="search-asset-type-sticker sticker sticker-default sticker-lg sticker-rounded sticker-static">
 							<svg class="lexicon-icon">
 								<use xlink:href="<%= searchResultSummaryDisplayContext.getPathThemeImages() %>/lexicon/icons.svg#<%= searchResultSummaryDisplayContext.getIconId() %>" />
+								<title>Icon</title>
```
Comment (carloslancha) [inline modules/apps/foundation/portal-search/portal-search-web/src/main/resources/META-INF/resources/search/results/view.jsp]: I guess title here should be <%= searchResultSummaryDisplayContext.getIconId() %>

**Comment by vbence86** [comment]
@carloslancha thanks for the head-ups. I'm addressing all your concerns now.

**Comment by vbence86** [comment]
@carloslancha I've substituted all hardcoded values with the most suitable text that was available in their current scope.

**Comment by carloslancha** [comment]
Nice @vbence86 thx!

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Pull request submitted to https://github.com/jbalsas/liferay-portal/pull/802. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Pull request submitted to https://github.com/jbalsas/liferay-portal/pull/803. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Pull request submitted to https://github.com/jbalsas/liferay-portal/pull/805. [See changes here.](https://github.com/carloslancha/liferay-portal/compare/0638349~1...0638349) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Pull request submitted to https://github.com/jbalsas/liferay-portal/pull/807. [See changes here.](https://github.com/carloslancha/liferay-portal/compare/0638349~1...0638349) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by carloslancha** [comment]
Pull request submitted to https://github.com/jbalsas/liferay-portal/pull/808. [See changes here.](https://github.com/carloslancha/liferay-portal/compare/b434532~2...b434532) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**